src/helpers/postMessageRelayHelpers.ts

```diff
@@ -10,14 +14,32 @@ import {
   EXPLORER_LISTENING_FOR_PARTIAL_TOKEN,
   PARENT_LOGOUT_SUCCESS,
   TRIGGER_LOGOUT_IN_PARENT,
+  EXPLORER_SUBSCRIPTION_RESPONSE,
+  EXPLORER_SET_SOCKET_ERROR,
+  EXPLORER_SET_SOCKET_STATUS,
 } from './constants';
-import type { JSONValue } from '../types';
+import type { JSONObject, JSONValue } from '../types';
+import { Client, createClient } from 'graphql-ws';
+import {
+  Observable,
+  OperationOptions,
+  SubscriptionClient as TransportSubscriptionClient,
+} from 'subscriptions-transport-ws';
 
 export type HandleRequest = (
   endpointUrl: string,
   options: Omit<RequestInit, 'headers'> & { headers: Record<string, string> }
 ) => Promise<Response>;
 
+export type HandleSubscription = (
+  params: OperationOptions & {
+    query: string;
+    variables: JSONObject;
+    operationName: string | undefined;
+  },
+  client: SubscriptionClient
```

Comment:
if someone is handling the subscription on their own, they are choosing their own client, so we shouldn't be setting up a `client` instance of a class at all.

Agh this makes this difficult because we don't want to have this line at all:

```
  const client = new SubscriptionClient(
    subscriptionUrl,
    headers ?? {},
    protocol
  );
```

if they have passed in their own handleSubscription.

How about we move the handleSubscription shtuff into a follow up PR and we can discuss there? I feel like we can definitely ship without it and follow up - it might require some more thinking.

Reply:
Basically I am thinking that we have no way of communicating back to folks that they have to close their subscription connection (EXPLORER_SUBSCRIPTION_TERMINATION), so I don't really know how this would work... We would also probably want to expose functions on the inputting `handleSubscription` for connecting, reconnecting etc. 

I think we might want to table `handleSubscription`. Lets at least split it out and talk to Chang about it when hes back

Reply:
Sounds good, I'll take the changes out and move to a new branch. I'll try to use the dev tools repo as a reference for the refactor.